integration-tests/tests/billing-projects.js

```diff
@@ -0,0 +1,92 @@
+const _ = require('lodash/fp')
+const { click, clickable, dismissNotifications, findText, noSpinnersAfter, select, signIntoTerra } = require('../utils/integration-utils')
+const { withUserToken } = require('../utils/terra-sa-utils')
+
+
+class BillingProjectsPage {
+  constructor(testPage, testUrl) {
+    this.testPage = testPage
+    this.testUrl = testUrl
+    this.mockValues = {}
+  }
+
+  async setAjaxMock(mockValues) {
+    this.mockValues = _.assign(this.mockValues, mockValues)
+    const spendReturnResult = {
+      spendSummary: {
+        cost: this.mockValues.spendCost, credits: '2.50', currency: 'USD', endTime: '2022-03-04T00:00:00.000Z', startTime: '2022-02-02T00:00:00.000Z'
+      }
+    }
+    const projectListResult = [{
+      projectName: this.mockValues.ownedBillingProjectName,
+      billingAccount: 'billingAccounts/fake-id', invalidBillingAccount: false, roles: ['Owner'], status: 'Ready'
+    }]
```

Comment:
This is the hazard that needs careful consideration. When the behavior of the target service changes, the confidence generated from this test should change, but I don't see anything that tells me we will become knowledgable of such an event.

Reply:
@dmohs That's true, but it's not a goal of the test. This is not an actual integration test (and perhaps we want to put these sort of tests in a different directory to make it clear). The goal of a test like this is to test the JavaScript code itself, so that we can be confident when we change utility methods or upgrade JavaScript 3rd party libraries, we have not broken anything. I explicitly do NOT want to be dependent on other services. I discuss this in the ticket itself-- https://broadworkbench.atlassian.net/browse/WOR-142

This test is actually a great example of why an integration test **cannot** work for this functionality. When getting the cost for the last X days, we depend on the BigQuery export of cost data from Google. There is no way I can keep actual data at an expected value from day-to-day, and depending on BigQuery data to exist introduces a ton of dependencies. I am trusting Rawls unit tests to enforce the contract from the API, and these tests are written against that contract. This type of test does not replace true integration tests, but it allows us to test _so many more things_, without the overhead and test flakiness we have seen with the existing integration tests.